aws-glue-datacatalog-client-common/src/main/java/com/amazonaws/glue/catalog/metastore/GlueMetastoreClientDelegate.java

```diff
@@ -189,6 +185,30 @@
   public static final String CATALOG_ID_CONF = "hive.metastore.glue.catalogid";
   public static final String NUM_PARTITION_SEGMENTS_CONF = "aws.glue.partition.num.segments";
 
+  private ExecutorService getExecutorService(HiveConf conf) throws MetaException {
+
+    String customFactoryClassName = conf.get(CUSTOM_EXECUTOR_FACTORY_CONF);
+    ExecutorService executorService = null;
+    if (!StringUtils.isEmpty(customFactoryClassName)) {
+      try {
+        Class<?> clazz = Class.forName(customFactoryClassName);
+        Object factory = clazz.newInstance();
+        Method method = clazz.getDeclaredMethod("getExecutorService", HiveConf.class);
+        executorService = (ExecutorService)method.invoke(factory, conf);
+      } catch(Exception e) {
+        logger.warn(e.getMessage());
+        throw new MetaException(e.getMessage() + e);
+      }
+    } else {
+        // If custom factory conf is not defined or if there is an error in loading it via reflection
```

Comment:
looks like you actually just throw an exception if there is an error loading via reflection  ?

Reply:
Fixed